content/guides/kubernetes/app-enhancements-probing-app-state.md

```diff
@@ -0,0 +1,199 @@
+---
+title:  "Probing Application State"
+parent: "Application Enhancements"
+topics:
+- Kubernetes
+tags:
+- Kubernetes
+team:
+- John Harris
+---
+
+Adding probes to your application provides two critical pieces of information to the system running
+it.
+
+1. Is the application ready to receive traffic?
+2. Is the application healthy?
+
+Cloud Native platforms have methods to probe the application and answer the above questions. In
+the case of Kubernetes, the kubelet (Kubernetes agent that runs on every host) can execute a 
+command inside the container, make an HTTP request, or make a TCP request.
+
+![kubelet-probe](/images/guides/kubernetes/app-enhancements/kubelet-probe.png)
+
+## Adding Probes to your Applications
+
+Two approaches for adding probes to your applications are:
+
+1. Add an endpoint that responds based on the state of the application.
+2. Run a command (process) in the container and validate condition or output.
+
+Option **1** is preferred. Since the endpoint is implemented in the application, it can perform an 
+in-depth check before responding to the requester. However, this option does require code 
+modification. An example of a simple Go application implementing this endpoint is as follows.
+
+```go
+package main
+
+import (
+	"log"
+	"net/http"
+)
+
+
+
+func main() {
+	// start the application routine
+	err := start()
+	if err != nil {
+		log.Panicln("failed to start.")
+	}
+
+  // TODO add http handler / server to validate health of app
+
+
+}
+
+func start() error {
+	// app logic for starting
+	return nil
+}
+
+func isHealthy() {
+	if 
```

Comment:
Incomplete example?

Reply:
Haha, weirdly this is like this internally as well. Guess we got distracted :joy:. Gonna remove it for now.